kvdb/cache.go

```diff
@@ -0,0 +1,930 @@
+package kvdb
+
+import (
+	"container/list"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/google/btree"
+)
+
+const (
+	treeDeg = 3
```

Comment:
Arbitrary value or was there some tuning here?

Reply:
Not much testing yet (I tested with 3 and 5) maybe best to change it to be a parameter instead.